### DISCUSSION TITLE
How could I call a function in app.go like that?

### DISCUSSION THREAD

**Discussion by godpowerscn** [wailsapp/wails#2313] [Q&A]
### Is your feature request related to a problem? Please describe.

Let me restate my requirements: I need to start a Timer or Ticker on main.go (background) and periodically send data to frontend (using EventOn). But I can't call A function like this(in main.go), how do I solve this problem?
 

func (m *MainStruct) TestMainFunc() {
     fmt.Println("This is TestMainFunc Action..." )
     runtime.EventsEmit(b.ctx, "TestEmitEvent", "text")
} 

func main(){
        // how cloud i call TestMainFunc here?
        ......
}

### Describe the solution you'd like

I looked for many references to solve this problem, but failed

**Comment by ZeronoFreya** [comment]
If I understand correctly, you need to use gorutine

Code is untested, please note:

```go
func main(){
   ticker := time.NewTicker(3 * time.Second)

    go func() {
        for {
            t := <-ticker.C
            runtime.EventsEmit(b.ctx, "TestEmitEvent", "text")
        }
    }()
}
```

**Reply by godpowerscn** [reply]
Thank you very much for your answer, but I still have some questions that I don't understand:

in the code : 

runtime.EventsEmit(b.ctx, "TestEmitEvent", "text")；

the code   "b.ctx"，where is b from?How and where do I define it?

**Reply by ZeronoFreya** [reply]
In app.go:
```go
func (a *App) startup(ctx context.Context) {
    // b is the a here, I don't know why your code uses b, but it's just another name
    a.ctx = ctx

    ticker := time.NewTicker(3 * time.Second)

    go func() {
        for {
            t := <-ticker.C
            runtime.EventsEmit(a.ctx, "TestEmitEvent", "text")
        }
    }()
}
This should work, I didn't test it.  good luck！
```
